### PR TITLE
SOFTWARE-5796: Fix ror id not populating in new institutions, present error info better in the frontend

### PR DESCRIPTION
**Pull request by mwestphall**

### CODE REVIEW

File: institutions-ui/src/app/institutions-editor/institutions-editor.component.ts [hunk 1]
```diff
@@ -35,10 +35,25 @@ export class InstitutionsEditorComponent implements OnInit {
...
 
+  extractErrorMessage(err: any) {
+    // Try to extract a meaningful error message out of the object returned by the server
+    return err?.error?.detail[0]?.msg || err?.error?.detail || err.message 
+  }
+
+  sanitizeFormData() {
+    // clean up form data prior to submission
+    let {name, id, ror_id } = this.institution
+    this.institution = { name: name.trim(), id: id.trim(), ror_id: ror_id?.trim() }
```
Comment (mwestphall) [inline institutions-ui/src/app/institutions-editor/institutions-editor.component.ts:51]: @brianhlin I believe the issue you saw with duplicate names was a case of one duplicate having a trailing space. This should sanitize leading/trailing whitespace from form data before submitting to the server, where the DB unique constraints will catch duplicates

Comment (brianhlin) [reply institutions-ui/src/app/institutions-editor/institutions-editor.component.ts:51]: Nice catch!

**Review by brianhlin — APPROVED** [review]
This LGTM but it looks like 3 logically separate changes in a single commit? Could you separate them out into different commits? Preapproving

File: institutions-api/db/db.py [hunk 2]
```diff
@@ -47,15 +52,38 @@ def get_institution(short_id: str) -> InstitutionModel:
...
-        if institution.ror_id:
-            inst.identifiers = [InstitutionIdentifier(_ror_id_type(session), institution.ror_id)]
         session.add(inst)
+        if institution.ror_id:
+            ror_id = InstitutionIdentifier(_ror_id_type(session), institution.ror_id, inst.id)
+            session.add(ror_id)
+
         session.commit()
 
+def _update_institution_ror_id(session: Session, institution: Institution, ror_id: str):
```
Comment (mwestphall) [inline institutions-api/db/db.py:67]: @brianhlin re: the discussion in https://github.com/osg-htc/topology-institutions-api/pull/1 about not always falling through to the delete-then-recreate workflow, I've split it out here into 3 explicit cases for delete, create and update

Comment (brianhlin) [reply institutions-api/db/db.py:67]: LGTM!

**Pull request merged**